Re: [patch 2/2, 7.3] Fix 7.1->7.2 regression: info sources
http://sourceware.org/ml/gdb-patches/2011-06/msg00146.html
http://sourceware.org/ml/gdb-cvs/2011-06/msg00056.html

### src/gdb/ChangeLog	2011/06/10 21:48:03	1.13087
### src/gdb/ChangeLog	2011/06/10 21:50:53	1.13088
## -1,5 +1,10 @@
 2011-06-10  Jan Kratochvil  <jan.kratochvil@redhat.com>
 
+	* symtab.c (output_partial_symbol_filename): Exchange the filename and
+	fullname parameters order.
+
+2011-06-10  Jan Kratochvil  <jan.kratochvil@redhat.com>
+
 	Code cleanup.
 	* dwarf2read.c (dw2_map_symbol_filenames): Use symbol_filename_ftype
 	for fun.
--- src/gdb/symtab.c	2011/05/18 16:30:36	1.274
+++ src/gdb/symtab.c	2011/06/10 21:50:54	1.275
@@ -2837,7 +2837,7 @@
 
 /* A callback for map_partial_symbol_filenames.  */
 static void
-output_partial_symbol_filename (const char *fullname, const char *filename,
+output_partial_symbol_filename (const char *filename, const char *fullname,
 				void *data)
 {
   output_source_filename (fullname ? fullname : filename, data);
### src/gdb/testsuite/ChangeLog	2011/06/10 16:21:46	1.2742
### src/gdb/testsuite/ChangeLog	2011/06/10 21:50:55	1.2743
## -1,3 +1,7 @@
+2011-06-10  Jan Kratochvil  <jan.kratochvil@redhat.com>
+
+	* gdb.dwarf2/dw2-filename.exp (info sources): New testcase.
+
 2011-06-10  Tom Tromey  <tromey@redhat.com>
 
 	* gdb.cp/temargs.exp: Let tests pass if compiler bug is fixed.
--- src/gdb/testsuite/gdb.dwarf2/dw2-filename.exp	2011/01/19 20:26:09	1.3
+++ src/gdb/testsuite/gdb.dwarf2/dw2-filename.exp	2011/06/10 21:50:55	1.4
@@ -38,3 +38,6 @@
 # is that the file and fullname fields are now inverted.
 gdb_test "interpreter-exec mi -file-list-exec-source-files" \
          ".*{file=\"file1\\.txt\",fullname=\".+file1\\.txt\"}.*"
+
+# And `info sources' should return the fullname incl. the directories.
+gdb_test "info sources" {[/\\]file1\.txt}
